### PR TITLE
Support Python 3.13

### LINKED ISSUE #2376: Support Python 3.13

### PR DESCRIPTION
**Pull request by sdrave**
- Adds support for Python 3.13 and tests it in CI
- Add threadpoolctl dependency to detect used BLAS/LAPACK in numpy, scipy
- Add `DEPENDENCY_INFO` attributes to `pymor.core.config` and print them in `pymor.core.config.__repr__`
- Report used BLAS/LAPACK library in `NUMPY_INFO`, `SCIPY_INFO`, etc.
- Make `lapack_driver` parameter for `scipy.linalg.svd` a pyMOR `default`.
- Choose `gesdd` if on Windows with MKL (see #2391).

Fixes #2376.

### CODE REVIEW

**Comment by sdrave** [comment]
Fixes #2376.

**Comment by HenKlei** [comment]
Ping due to soft freeze @sdrave.

File: .github/actions/miniconda_tests/action.yml [hunk 1]
```diff
@@ -82,6 +82,12 @@ runs:
         ${PYTEST_PREFIX} python -c "from qtpy.QtGui import *"
         ${PYTEST_PREFIX} python -c "import OpenGL"
         ${PYTEST_PREFIX} python -c "import OpenGL.GL as gl"
+    - name: Print config
+      shell: bash -el {0}
+      env:
+        PYTHONPATH: ./src
+      run: |
+        ${PYTEST_PREFIX} python -c "import pymor; print(pymor.config)"
```
Comment (HenKlei) [inline .github/actions/miniconda_tests/action.yml:90]: Should we leave that in here or was it just for testing purposes?

File: src/pymor/bindings/scipy.py [hunk 2]
```diff
@@ -13,14 +13,33 @@
...
+@defaults('driver')
+def svd_lapack_driver(driver='gesvd_unless_win_mkl'):
+    assert driver in {'gesvd', 'gesdd', 'gesvd_unless_win_mkl'}
+    if driver == 'gesvd_unless_win_mkl':
+        if is_windows_platform() and is_scipy_mkl():
+            from warnings import warn
+            warn('Your SciPy installation seems to be using MKL as LAPACK library.\n'
+                 'To avoid stability issues on Windows, we use gesdd instead of gesvd\n'
+                 'for SVD computation. This may lead to reduced numerical accuracy.\n'
+                 'See https://github.com/pymor/pymor/issues/2391 for further discuson.\n'
```
Comment (HenKlei) [inline src/pymor/bindings/scipy.py:34]:
```suggestion
                 'See https://github.com/pymor/pymor/issues/2391 for further discussion.\n'
```

**Review by HenKlei — APPROVED** [review]
Looks good to me.

File: .github/actions/miniconda_tests/action.yml (hunk 1 above)
Comment (sdrave) [reply .github/actions/miniconda_tests/action.yml:90]: Yes, it makes sense to keep that.

**Pull request merged**